### PR TITLE
Testing/initial tests

### LINKED ISSUE #3: create tests for `Stat`
As a developer, 
When I see a `Stat` component, 
I want to see testing for the component,
So that I know the component does what I expect it do do.

### LINKED ISSUE #4: create tests for `ComparisonContainer`
As a developer, 
When I see a `ComparisonContainer` component, 
I want to see testing for the component,
So that I know the component does what I expect it do do.

### LINKED ISSUE #5: create tests for `USAState`
As a developer, 
When I see a `USAState` component, 
I want to see testing for the component,
So that I know the component does what I expect it do do.

### LINKED ISSUE #6: create tests for `ComparisonDetails`
As a developer, 
When I see a `ComparisonDetails` component, 
I want to see testing for the component,
So that I know the component does what I expect it do do.

### PR DESCRIPTION
**Pull request by kncaputo**
## What's this PR do?
- Creates render tests for `App`, `USAState`, `Stat`, `ComparisonContainer`
- Add format date method to `App` (this was necessary for `USAState` testing)

## Where should the reviewer start?
- `App.test`, `USAState.test`, `Stat.test`, `ComparisonContainer.test`, `formatDate` in `App`

## How should this be manually tested?
- run test suite, all tests should pass
- in browser, on page load, date should be formatted '01/01/2021'

### CODE REVIEW

File: src/App/App.test.tsx [hunk 1]
```diff
@@ -1,9 +1,49 @@
...
+import { render, screen, fireEvent } from '@testing-library/react';
 import App from './App';
+import { fetchAllCurrentUSAData, fetchCurrentStateData } from '../apiCalls'
+import { singleState, usaData } from '../sampleData';
+import '@testing-library/jest-dom';
+jest.mock('../apiCalls');
+ 
+describe('App', () => {
+  beforeEach(() => {
+    const mockedUSAFetch = fetchAllCurrentUSAData as jest.MockedFunction<typeof fetchAllCurrentUSAData>
```
Comment (alyssajoyner) [inline src/App/App.test.tsx:10]: Great work figuring mockFetch in TypeScript!

File: src/App/App.test.tsx [hunk 2]
```diff
@@ -1,9 +1,49 @@
...
+    expect(stateName).toBeInTheDocument();
+    expect(cases).toBeInTheDocument();
+    expect(deaths).toBeInTheDocument();
+    expect(currentHospitalizations).toBeInTheDocument();
+    expect(howDoesItCompare).toBeInTheDocument();
+    expect(dropdown).toBeInTheDocument();
+    expect(casesIcon).toBeInTheDocument();
+  });
+
+  it('should simplify data for single state', () => {
```
Comment (alyssajoyner) [inline src/App/App.test.tsx:43]: We should reach out to a mentor/instructor on the best way to handle testing chained functions in API calls, the docs were skimpy.

File: src/App/App.test.tsx [hunk 3]
```diff
@@ -1,9 +1,49 @@
...
 import App from './App';
+import { fetchAllCurrentUSAData, fetchCurrentStateData } from '../apiCalls'
+import { singleState, usaData } from '../sampleData';
+import '@testing-library/jest-dom';
+jest.mock('../apiCalls');
+ 
+describe('App', () => {
+  beforeEach(() => {
+    const mockedUSAFetch = fetchAllCurrentUSAData as jest.MockedFunction<typeof fetchAllCurrentUSAData>
+    const mockedSingleFetch = fetchCurrentStateData as jest.MockedFunction<typeof fetchCurrentStateData>
```
Comment (Codo-Baggins) [inline src/App/App.test.tsx:11]: Nice job mocking these correctly!

File: src/App/App.test.tsx [hunk 4]
```diff
@@ -1,9 +1,49 @@
...
+  it('should render correctly', () => {
+    const tagline = screen.getByText('Covid Comparisons');
+    const overviewTitle = screen.getByText('USA Overview:');
+    const stateName = screen.getByText('Colorado');
+    const cases = screen.getByText('Cases');
+    const deaths = screen.getByText('Deaths');
+    const currentHospitalizations = screen.getByText('Current Hospitalizations');
+    const howDoesItCompare = screen.getByText('How Does It Compare?');
+    const dropdown = screen.getByTestId('dropdown');
+    const casesIcon = screen.getByTestId('cases');
```
Comment (alyssajoyner) [inline src/App/App.test.tsx:30]: Might be worth adding a test to ensure that 3 icons are rendered on the page.

File: src/App/App.test.tsx [hunk 5]
```diff
@@ -1,9 +1,49 @@
...
+    
+    expect(tagline).toBeInTheDocument();
+    expect(overviewTitle).toBeInTheDocument();
+    expect(stateName).toBeInTheDocument();
+    expect(cases).toBeInTheDocument();
+    expect(deaths).toBeInTheDocument();
+    expect(currentHospitalizations).toBeInTheDocument();
+    expect(howDoesItCompare).toBeInTheDocument();
+    expect(dropdown).toBeInTheDocument();
+    expect(casesIcon).toBeInTheDocument();
```
Comment (Codo-Baggins) [inline src/App/App.test.tsx:40]: Should some of these be broken out into the tests for their individual components?  Maybe something we should revisit in the near future.

**Review by alyssajoyner — APPROVED** [review]
Nice start to testing - this gives us a great foundation to add integration testing one the comparison details component is built out.

File: src/App/App.test.tsx [hunk 6]
```diff
@@ -1,9 +1,49 @@
...
+  });
+
+  it('should simplify data for single state', () => {
+    // const simplifyAPIDataForSingleState = jest.fn();
+    
+    // expect(simplifyAPIDataForSingleState).toHaveBeenCalled();
+    // expect(mockSimplifyAPIDataForSingleState).toBeCalledWith(singleState);
+  });
```
Comment (Codo-Baggins) [inline src/App/App.test.tsx:48]: Still trying to determine if we need to test for those functions in our App.tsx file and if so, how we would test them in the first place.

**Review by Codo-Baggins — APPROVED** [review]
This is a good start! As we go we can add more tests!

**Pull request merged**